solutions/poi-04-spies.mdx

```diff
@@ -262,7 +262,10 @@ int main() {
 }
 ```
 
-OK:
+To avoid this, we can make a few memory optimizations.
+
+1. Since this is a functional graph, each node is directed towards one other node. Instead of `vector`, we can use an `int` array to store each node's successor as the adjacency list.
```

Comment:
^ it's not clear what this means

Reply:
suggest something like:

Since this is a functional graph, each node has exactly one edge leaving it. Therefore, instead of representing the graph as an adjacency list, we can represent it as an array where arr[i] stores f(i).

Reply:
that doesn't explain what `ST` and `EN` are for though

Reply:
oh didn't notice that. are ST and EN meant for some cycle detection stuff?